src/firefly/js/util/Interp/Regrid.js

```diff
@@ -0,0 +1,95 @@
+
+import {LinearInterpolator} from './LinearInterpolator';
+/**
+ * The Regrid class shrinks or expands the size of an array by an
+ * arbitrary amount.  This class is similar to "Rebin" that we was not implemented in that it can
+ * resize a one, two, or three dimensional array.  "Rebin", however, requires that the new array size must be an
+ * integer multiple of the original size.  Regrid will resize an array to any arbitrary size
+ * (Rebin is somewhat faster, however).  Rebin averages multiple points when shrinking an array, while Regrid just
+ * resamples the array. The returned array has the same number of dimensions as the original
+ * array and is of the same data type.
+ *
+ *
+ * NOTE: this implementation is one dimension only.
+ *
+ *  The algorithm is the same as IDL CONGRID with MINUS_ONE is set to true and extropolation=false by default
+ *
+ *        When MINUS_ONE is true, I enforced the new coordinate at newDim -1 is
+ *          the same as oldDim-1.   Mathematically,
+ *             RegridFact = (oldDim - 1) / (newDim - 1);
+ *             new_coordinate(i) =   i * RegridFact
+ *             when i = newDim -1, new_coordinate(newDim-1) = oldDim -1.
+ *             But due to the numerically error, there may be some small detla value introduced.
+ *             Therefore, I need to enforce it to oldDim -1
+ *
+ */
+
+export function Regrid(inData, outSize, allowExtrapolation=false){
+
+
+    var outData = new Array(outSize);
+
+    const outDataCoord = generateNewCoordinates(inData, outSize, allowExtrapolation);
+
+    const inCoords = Array.from(new Array(inData.length), (x,i) => i);
+    const linearInterpolator = new LinearInterpolator(inCoords, inData, allowExtrapolation);
+
+    for (var i=0; i<outSize; i++){
+        outData[i]=linearInterpolator.getInterpolatedValue(outDataCoord[i]);
+    }
```

Comment:
Are you doing a `for` loop to get better performance with a large array. Otherwise you might want to use map.
```
const outData= outDataCoord.map( (c) => linearInterpolator.getInterpolatedValue(c));
```